src/sdk/v2/sdk.ts

```diff
@@ -93,11 +122,16 @@ const _checkSignerInitialized = (state: SdkStore, fnName: string) => {
  *
  * @param {PermitV2Options} options - Partial PermitV2 fields to create the Permit with, if no options provided will be filled with the defaults:
  * { type: "self", issuer: initializedUserAddress, projects: initializedProjects, contracts: initializedContracts }
- * @returns {PermitV2} - Newly created PermitV2
+ * @returns {Result<PermitV2>} - Newly created PermitV2 as a Result object
  */
-const createPermit = async (options?: PermitV2Options): Promise<PermitV2> => {
+const createPermit = async (
+  options?: PermitV2Options,
+): Promise<Result<PermitV2>> => {
   const state = _sdkStore.getState();
-  _checkSignerInitialized(state, createPermit.name);
+
+  const initialized = _checkInitialized(state);
+  if (!initialized.success)
+    return ResultErr(`${createPermit.name} :: ${initialized.error}`);
```

Comment:
A lot of the changes below are switching the return type of the sdk functions from `<value>` to `Result<value>`. This means that error messages are returned as part of the functions return type, instead of being thrown as a base JS Error. JS Errors are difficult to deal with in a client and require a lot of error handling and boundaries. The standard now is to return errors as part of a return object (see ReactQuery).